library/ssl_tls.c

```diff
@@ -1940,6 +1953,12 @@ static int ssl_tls13_session_save( const mbedtls_ssl_session *session,
                                    size_t *olen )
 {
     unsigned char *p = buf;
+#if defined(MBEDTLS_SSL_CLI_C) && \
+    defined(MBEDTLS_SSL_SESSION_TICKETS) && \
+    defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
+    size_t hostname_len = ( session->hostname == NULL ) ?
+                            0 : strlen( session->hostname );
```

Comment:
hostname_len should be the size needed.
```suggestion
    size_t hostname_len = ( session->hostname == NULL ) ?
                            0 : strlen( session->hostname ) + 1;
```

Reply:
I am not sure whether there is empty servername only has "\0", if there is not such one, maybe we have no need to wast one extra byte space.
Did someone learn about such cases?

Reply:
`session->hostname` is come from `ssl->hostname` . And `ssl->hostname` is set by `mbedtls_ssl_set_hostname`.  it allows empty string.